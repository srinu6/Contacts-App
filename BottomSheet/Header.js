import React, { useState } from "react";
import {View,TextInput,Text,Alert,TouchableOpacity,ImageBackground,StyleSheet} from 'react-native';
import ImagePicker from 'react-native-image-crop-picker';
import Animated from 'react-native-reanimated';
import BottomSheetOriginal from 'reanimated-bottom-sheet';
import {choosePhotoFromLibrary, takePhotoFromCamera} from '../components/AddContact'

const bottomsheet = React.createRef();
// let req=true;
// export const FromCamera=()=>{
  
//   return true;
// }
//{req == true? false: null}
// export const FromLibrary=()=>{
// if(!req){
//   return true;
// }
// return false;
// }



  const styles = StyleSheet.create({
    nameTextInputStyle: {
      height: 40, 
      borderColor: 'gray', 
      borderWidth: 3, 
      borderRadius:15, 
      flex:1, 
      padding:10  
    },
    phoneEmailTextInputStyle: {
      height: 40, 
      borderColor: 'gray', 
      borderWidth: 3, 
      marginTop:10,
      borderRadius:15, 
      padding:10  
    },
    fullNameRow:{ 
      flexDirection: "row", 
      marginTop:20 
    },
    errorText:{
      color:'red' 
    },
    photoText: {
      marginTop: 10, 
      fontSize: 18, 
      fontWeight: 'bold'
    },
    photoIcon: {
      opacity: 0.7,
      alignItems: 'center',
      justifyContent: 'center',
      borderWidth: 1,
      borderColor: '#fff',
      borderRadius: 10,
    },
    photoIconView: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
    },
    imageBackGround: {
      height: 100, 
      width: 100, 
      backgroundColor:'#666b68', 
      borderRadius: 45
    },
    imageBackGroundView: {
      height: 100,
      width: 100,
      borderRadius: 15,
      justifyContent: 'center',
      alignItems: 'center',
    },
    
    commandButton: {
      padding: 10,
      borderRadius: 10,
      backgroundColor: '#FF6347',
      alignItems: 'center',
      marginTop: 10,
    },
    panel: {
      padding: 20,
      backgroundColor: '#FFFFFF',
      paddingTop: 20,
    },
    header: {
      backgroundColor: '#FFFFFF',
      shadowColor: '#333333',
      shadowOffset: {width: -1, height: -3},
      shadowRadius: 2,
      shadowOpacity: 0.4,
      paddingTop: 20,
      borderTopLeftRadius: 20,
      borderTopRightRadius: 20,
    },
    panelHeader: {
      alignItems: 'center',
    },
    panelHandle: {
      width: 40,
      height: 8,
      borderRadius: 4,
      backgroundColor: '#00000040',
      marginBottom: 10,
    },
    panelTitle: {
      fontSize: 27,
      height: 35,
    },
    panelSubtitle: {
      fontSize: 14,
      color: 'gray',
      height: 30,
      marginBottom: 10,
    },
    panelButton: {
      padding: 13,
      borderRadius: 10,
      backgroundColor: '#FF6347',
      alignItems: 'center',
      marginVertical: 7,
    },
    panelButtonTitle: {
      fontSize: 17,
      fontWeight: 'bold',
      color: 'white',
    },
  });