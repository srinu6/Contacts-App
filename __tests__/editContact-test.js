import 'react-native';
import React from 'react';
import AddEditContact from '../components/addEditContacts';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {mount, shallow} from 'enzyme';
import {render, fireEvent} from 'react-native-testing-library';
import BottomSheet from 'reanimated-bottom-sheet';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));
jest.mock('react-native-image-crop-picker', () => ({
  openCamera: jest.fn().mockImplementation(() => Promise.resolve()),
  openPicker: jest.fn().mockImplementation(() => Promise.resolve()),
}));

jest.mock('reanimated-bottom-sheet');
jest.mock('react-native-reanimated', () => {
  const View = require('react-native').View;
  return {
    Value: jest.fn(),
    event: jest.fn(),
    add: jest.fn(),
    eq: jest.fn(),
    set: jest.fn(),
    cond: jest.fn(),
    interpolate: jest.fn(),
    multiply: jest.fn(),
    View: View,
    Extrapolate: {CLAMP: jest.fn()},
    Transition: {
      Together: 'Together',
      Out: 'Out',
      In: 'In',
    },
  };
});

const navigation = {navigate: jest.fn()};
const routeForEditContact = {
  params: {
    contactId: 1,
    addorEdit: false,
  },
};
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  contactStore: {
    contact: [
      {
        id: 0,
        firstName: 'Iron',
        lastName: 'Man',
        email: 'ironman@hero.com',
        phone: '9848022335',
        image: null,
      },
      {
        id: 1,
        firstName: 'Super',
        lastName: 'Man',
        email: 'superman@hero.com',
        phone: '9876543234',
        image: null,
      },
      {
        id: 2,
        firstName: 'Thor',
        lastName: 'God',
        email: 'Thor@hero.com',
        phone: '1234567898',
        image: null,
      },
      {
        id: 3,
        firstName: 'Thanos',
        lastName: 'Conqurer',
        email: 'Thanos@powerfull.com',
        phone: '5432145678',
        image: null,
      },
    ],
  },
});

const storeConditionFirstName = mockStore({
  contactStore: {
    contact: [
      {
        id: 0,
        firstName: 'Iron',
        lastName: 'Man',
        email: 'ironman@hero.com',
        phone: '9848022335',
        image: null,
      },
      {
        id: 1,
        firstName: 'Super3',
        lastName: 'Man',
        email: 'superman@hero.com',
        phone: '9876543234',
        image: null,
      },
    ],
  },
});

const storeConditionLastName = mockStore({
  contactStore: {
    contact: [
      {
        id: 0,
        firstName: 'Iron',
        lastName: 'Man',
        email: 'ironman@hero.com',
        phone: '9848022335',
        image: null,
      },
      {
        id: 1,
        firstName: 'Super',
        lastName: 'Man1',
        email: 'superman@hero.com',
        phone: '9876543234',
        image: null,
      },
    ],
  },
});

const storeConditionPhoneNumber = mockStore({
  contactStore: {
    contact: [
      {
        id: 0,
        firstName: 'Iron',
        lastName: 'Man',
        email: 'ironman@hero.com',
        phone: '9848022335',
        image: null,
      },
      {
        id: 1,
        firstName: 'Super',
        lastName: 'Man',
        email: 'superman@hero.com',
        phone: '9876543234f',
        image: null,
      },
    ],
  },
});

const storeConditionEmail = mockStore({
  contactStore: {
    contact: [
      {
        id: 0,
        firstName: 'Iron',
        lastName: 'Man',
        email: 'ironman@hero.com',
        phone: '9848022335',
        image: null,
      },
      {
        id: 1,
        firstName: 'Super',
        lastName: 'Man',
        email: 'superman@herocom',
        phone: '9876543234',
        image: null,
      },
    ],
  },
});


describe('Should call Edit Contact', () => {
  it('Updating a Contact, this will check Update Contact', () => {
    const {queryByText} = render(
      <AddEditContact route={routeForEditContact} navigation={navigation} />,
    );
    expect(queryByText('Update Contact')).not.toBeNull();
  });
});

describe('Should call Edit Contact for checking TestInput', () => {
  it('change input First Name, Last Name, Phone Number, EmailID', () => {
    const {getByTestId} = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );

    fireEvent.changeText(getByTestId('firstName'), 'Srinu');
    expect(getByTestId('firstName').props.value).toEqual('Srinu');

    fireEvent.changeText(getByTestId('lastName'), 'Maripi');
    expect(getByTestId('lastName').props.value).toEqual('Maripi');

    fireEvent.changeText(getByTestId('phoneNumber'), '0123456789');
    expect(getByTestId('phoneNumber').props.value).toEqual('0123456789');

    fireEvent.changeText(getByTestId('emailId'), 'srinu@gmail.com');
    expect(getByTestId('emailId').props.value).toEqual('srinu@gmail.com');
  });
});

describe('Checks different TouchableOpacities', () => {
  it('Update & Navigation touchable opacity', () => {
    const componentTouch = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    const touchableSubmit = componentTouch.getByTestId('submit');
    fireEvent.press(touchableSubmit);

    const touchableNavigation = componentTouch.getByTestId('navigation');
    fireEvent.press(touchableNavigation);
  });
});

describe('Checking submit TouchableOpacity, conditions, validFirstNameCheck', () => {
  it('if else condition checking ', () => {
    const componentTouch = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={storeConditionFirstName}
      />,
    );
    const touchableSubmit = componentTouch.getByTestId('submit');
    fireEvent.press(touchableSubmit);
  });
});

describe('Checking submit TouchableOpacity, conditions, validLastNameCheck', () => {
  it('if else condition checking, validLastNameCheck ', () => {
    const componentTouch = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={storeConditionLastName}
      />,
    );
    const touchableSubmit = componentTouch.getByTestId('submit');
    fireEvent.press(touchableSubmit);
  });
});

describe('Checking submit TouchableOpacity, conditions, validPhoneCheck', () => {
  it('if else condition checking, validPhoneCheck ', () => {
    const componentTouch = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={storeConditionPhoneNumber}
      />,
    );
    const touchableSubmit = componentTouch.getByTestId('submit');
    fireEvent.press(touchableSubmit);
  });
});

describe('Checking submit TouchableOpacity, conditions, validEmailCheck', () => {
  it('if else condition checking, validEmailCheck ', () => {
    const componentTouch = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={storeConditionEmail}
      />,
    );
    const touchableSubmit = componentTouch.getByTestId('submit');
    fireEvent.press(touchableSubmit);
  });
});

describe('Should call Edit Contact, to check branch', () => {
  it('useSelector branch checking', () => {
    const wrapperMount = mount(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={null}
      />,
    );
    console.log(wrapperMount, 'useSelector branch');
  });
});

describe('Should call Edit Contact', () => {
  it('this will check name field in Edit Contact', () => {
    const {getByTestId} = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    expect(getByTestId('firstName').props.value).toEqual('Super');
    expect(getByTestId('lastName').props.value).toEqual('Man');
  });
});

describe('Should call Edit Contact', () => {
  it('this will check phone numeber field in Edit Contact', () => {
    const {getByTestId} = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    expect(getByTestId('phoneNumber').props.value).toEqual('9876543234');
  });
});

describe('Should call Edit Contact', () => {
  it('this will check Email ID field in Edit Contact', () => {
    const {getByTestId} = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    expect(getByTestId('emailId').props.value).toEqual('superman@hero.com');
  });
});

describe('Should call Edit Contact', () => {
  it('onBlur First Name, Last Name, Phone Number, EmailID', () => {
    const {getByTestId} = render(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );

    fireEvent(getByTestId('firstName'), 'blur');
    fireEvent(getByTestId('lastName'), 'blur');
    fireEvent(getByTestId('phoneNumber'), 'blur');
    fireEvent(getByTestId('emailId'), 'blur');
  });
});

describe('Should call Edit Contact', () => {
  it('functions calling, RenderHeader', () => {
    const wrapshallow = shallow(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    const bottomSheetFind = wrapshallow.find(BottomSheet);
    const renderHeaderPart = bottomSheetFind.renderProp('renderHeader')();
    expect(renderHeaderPart).toMatchSnapshot();
  });
});

describe('Should call Edit Contact', () => {
  it('functions calling, RenderInner', () => {
    const wrapshallowinner = shallow(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    const bottomSheetFind = wrapshallowinner.find(BottomSheet);
    const renderInnerPart = bottomSheetFind.renderProp('renderContent')();
    expect(renderInnerPart).toMatchSnapshot();
  });
});

describe('Should call Edit Contact', () => {
  it('functions calling, RenderInner, not a snap', () => {
    const wrapShallowInner = shallow(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    const bottomSheetFind = wrapShallowInner.find(BottomSheet);
    const renderInnerPart = bottomSheetFind.renderProp('renderContent')();
    const {getByTestId} = render(renderInnerPart);
    fireEvent.press(getByTestId('uploadphoto'));
  });
});

describe('Should call Edit Contact & choosePhotoFromLibrary function', () => {
  it('functions calling, RenderInner, choosePhotoFromLibrary', () => {
    const wrapShallowHeader = shallow(
      <AddEditContact
        route={routeForEditContact}
        navigation={navigation}
        store={store}
      />,
    );
    const bottomSheetFindHeader = wrapShallowHeader.find(BottomSheet);
    const renderHeaderPart = bottomSheetFindHeader.renderProp(
      'renderContent',
    )();
    const {getByTestId} = render(renderHeaderPart);
    fireEvent.press(getByTestId('libraryphoto'));
  });
});
