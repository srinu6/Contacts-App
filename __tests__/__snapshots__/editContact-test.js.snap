// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call Edit Contact functions calling, RenderHeader 1`] = `ShallowWrapper {}`;

exports[`Should call Edit Contact functions calling, RenderInner 1`] = `ShallowWrapper {}`;
